apps/storefront/src/pages/QuoteDraft/index.test.tsx

```diff
@@ -1331,431 +1331,607 @@ describe('when the user is a B2B customer', () => {
+
+        expect(within(productTable).getAllByText('TBD')).toHaveLength(2);
       });
 
-      const productTable = await screen.findByRole('table');
+      it('does not show TBD as price when unlimited backorder is true', async () => {
+        const product = buildDraftQuoteItemWith({
+          node: {
+            quantity: 10,
+            variantSku: 'LC-123',
+            productsSearch: buildProductWith({
+              availability: 'available',
+              inventoryLevel: 0,
+              availableToSell: 5,
+              unlimitedBackorder: true,
+              inventoryTracking: 'product',
+              variants: [
+                buildVariantWith({
+                  purchasing_disabled: false,
+                  sku: 'LC-123',
+                }),
+              ],
+            }),
+          },
+        });
 
-      expect(within(productTable).queryByText('Insufficient stock')).not.toBeInTheDocument();
-    });
+        const quoteInfo = buildQuoteInfoStateWith({
+          draftQuoteInfo: {
+            contactInfo: { email: customerEmail },
+            billingAddress: noAddress,
+            shippingAddress: noAddress,
+          },
+          draftQuoteList: [product],
+        });
 
-    it('shows TBD as price when available to sell is insufficient', async () => {
-      const product = buildDraftQuoteItemWith({
-        node: {
-          quantity: 10,
-          variantSku: 'LC-123',
-          productsSearch: buildProductWith({
-            availability: 'available',
-            inventoryLevel: 0,
-            availableToSell: 5,
-            unlimitedBackorder: false,
-            inventoryTracking: 'product',
-            variants: [
-              buildVariantWith({
-                purchasing_disabled: false,
-                sku: 'LC-123',
-              }),
-            ],
-          }),
-        },
-      });
+        renderWithProviders(<QuoteDraft setOpenPage={vi.fn()} />, {
+          preloadedState: {
+            ...preloadedState,
+            quoteInfo,
+            global: buildGlobalStateWith({
+              blockPendingQuoteNonPurchasableOOS: { isEnableProduct: true },
+              featureFlags,
+            }),
+          },
+        });
 
-      const quoteInfo = buildQuoteInfoStateWith({
-        draftQuoteInfo: {
-          contactInfo: { email: customerEmail },
-          billingAddress: noAddress,
-          shippingAddress: noAddress,
-        },
-        draftQuoteList: [product],
-      });
+        const productTable = await screen.findByRole('table');
 
-      renderWithProviders(<QuoteDraft setOpenPage={vi.fn()} />, {
-        preloadedState: {
-          ...preloadedState,
-          quoteInfo,
-          global: buildGlobalStateWith({
-            blockPendingQuoteNonPurchasableOOS: { isEnableProduct: true },
-            featureFlags,
-          }),
-        },
+        expect(within(productTable).queryAllByText('TBD')).toHaveLength(0);
```

Comment:
🤔 should this be the opposite? check that the price is present?

I'm ok if we want ot make sure TBD is not present, but I feel we might want to ensure the prices are shown.


```tsx
it('shows the price when unlimited backorder is true', async () => {
```